drivers/serial/uart_stm32.c

```diff
@@ -336,10 +336,10 @@ static int uart_stm32_configure(const struct device *dev,
 	/* Driver doesn't support 5 or 6 databits and potentially 7 or 9 */
 	if ((UART_CFG_DATA_BITS_5 == cfg->data_bits) ||
 	    (UART_CFG_DATA_BITS_6 == cfg->data_bits)
-#ifndef LL_USART_DATAWIDTH_7B
+#ifdef LL_USART_DATAWIDTH_7B
 	    || (UART_CFG_DATA_BITS_7 == cfg->data_bits)
 #endif /* LL_USART_DATAWIDTH_7B */
-#ifndef LL_USART_DATAWIDTH_9B
+#ifdef LL_USART_DATAWIDTH_9B
```

Comment:
I propose the #ifdef is removed to fit conclusion of discussions on https://github.com/zephyrproject-rtos/zephyr/issues/31799

Reply:
I have removed the ifdefs.